domain/port/watcher_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package port_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/watcher/watchertest"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/application"
+	"github.com/juju/juju/domain/application/charm"
+	applicationerrors "github.com/juju/juju/domain/application/errors"
+	applicationstate "github.com/juju/juju/domain/application/state"
+	machinestate "github.com/juju/juju/domain/machine/state"
+	"github.com/juju/juju/domain/port/service"
+	"github.com/juju/juju/domain/port/state"
+	changestreamtesting "github.com/juju/juju/internal/changestream/testing"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/logger"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+)
+
+type watcherSuite struct {
+	changestreamtesting.ModelSuite
+
+	srv *service.WatchableService
+
+	unitCount int
+
+	unit0UUID string
+	unit1UUID string
+	unit2UUID string
+
+	app0UUID string
+	app1UUID string
+}
+
+var _ = gc.Suite(&watcherSuite{})
+
+var (
+	ssh   = network.PortRange{Protocol: "tcp", FromPort: 22, ToPort: 22}
+	http  = network.PortRange{Protocol: "tcp", FromPort: 80, ToPort: 80}
+	https = network.PortRange{Protocol: "tcp", FromPort: 443, ToPort: 443}
+)
+
+const (
+	m0UUID       = "machine-0-uuid"
+	m1UUID       = "machine-1-uuid"
+	netNode0UUID = "net-node-0-uuid"
```

Comment:
**nitpick:** numeric values in the middle of a variable is not exactly readable.

suggestion: Use a slice, even if it make them var, or put the numeric at the end of the name.